privilege-request-protocol.md

```diff
@@ -0,0 +1,208 @@
+guarantee that sufficient privilege has been granted at this point, and new
+requests might still be denied, potentially restarting this process.
+
+Syntax
+------
+The link to the privilege request endpoint is in an HTTP `Link` response
+header of the original denied transaction. The link relation is *TBD*, but for
+now let us use `x-permission-request` as a placeholder pending an agreeable
+identifier. The link target URI is opaque to the client, need not be for the
+same origin as the original denied request, **SHOULD** be unguessable and
+unforgeable, and **SHOULD** only be valid for a limited time and a single
+use.  The privilege request endpoint **MUST** be able to sufficiently identify
+the client solely by this URI, for example by the URI comprising a key to a
+database record, or by comprising a signed serialization of relevant information
+about the client.  The `Link` header **SHOULD** include an `expires_in`
+parameter giving the number of seconds after the `Date` of this response at
+which the privilege request URI will no longer be valid.
+
+A request for additional privilege **SHALL** be made by HTTP `POST` to the
+privilege request URI. The endpoint **MUST NOT** require any special credentials,
+including any authentication or authorization credentials included with the
+original request; that is, the URI itself is a sufficient [capability][]
+for this specific request.
+
+The request body of the `POST` to the privilege request endpoint, if present,
+**SHALL** have a content-type of `application/x-www-form-urlencoded`. At this
+time no parameters are defined. A possible future extension could include a
+[webhook][] to be notified on completion of the management activity. The
+privilege request endoint **MUST** ignore unrecognized parameters.
+
+A successful response (`2XX`) to a request for additional privilege indicates
+that the request was accepted, not necessarily that it was completed or that
+sufficient privilege has been or will be granted. A successful response
+**SHOULD** use response code `202 Accepted`.
+
+A successful response **MAY** include a response body. If included, the
+response body **MUST** be in `application/json` format encoding a JSON
+object. The following object key is defined:
+
+* *Key TBD* (**OPTIONAL**): a [Level 1 URI template][RFC6570] string for a web-based
+  permission management interface. For now let us use `x-permission-management-page`
+  as a placeholder key pending an agreeable identifier.  If followed by the
+  client, the URI **SHOULD** be opened in the user's trusted web browser, and not
+  in a frame. The web-based permission management interface **SHOULD** use
+  frame-busting techniques and **SHOULD** take care to defeat cross-site
+  request forgery, click-jacking, and similar attacks.  The URI can include
+  the following template variable:
+
+  * `redirect_uri` (**RECOMMENDED**): A target to resume the client application.
```

Comment:
maybe instead of `redirect_uri` use something that will not normally appear in the rest of the URL, `${redirect_uri}` or REDIRECT_URI instead, so that a template could be for instance:
https://example.com/request?redirect_uri=REDIRECT_URI

Reply:
the permission management page link uses an [RFC 6570 Level 1 URL template](https://tools.ietf.org/html/rfc6570). the template pattern is `{redirect_uri}`, so in your example would be `https://example.com/request?redirect_uri={redirect_uri}`. there's no concern with a collision or conflict.  the entry here is talking about the name of the template variable, in the terminology of RFC 6570.

Reply:
ah ok